### ISSUE TITLE
Compatibility with Turing models

### ISSUE THREAD

**Issue by DominiqueMakowski** [beacon-biosignals/Effects.jl#76]
Thanks forthe package! 
Any plans to make Effects.jl work with (at least some simple) Turing models? 
What would be the things needed for it to work?

**Comment by palday** [comment]
This might be doable as a package extension. The main problem is that AFAIK Turing models don't support the `StatsAPI.RegressionModel` interface. The main functions we depend on are:

- `modelmatrix`
- `coef`
- `formula` (to extract the model formula and then use that to generate typical values for marginalized terms)
- `vcov`

I suspect that you would actually want to not use `vcov` directly, but rather do something more like we do in the [experimental bootstrap support](https://github.com/beacon-biosignals/Effects.jl/blob/7771901d5263f5eba95ecdee50599c1ea97bc719/ext/EffectsMixedModelsExt.jl#L36-L94) in the MixedModels extension, but using posterior draws instead of bootstrap replicates.

**Comment by DominiqueMakowski** [comment]
Indeed, maybe a good start to add the above methods would be in TuringGLM falls more into a standard regression type.
